enhancements/automated-resource-request-scaling.md

```diff
@@ -0,0 +1,517 @@
+not themselves.
+
+### Locations for API setting
+
+These alternatives could apply to any of the API forms discussed in
+the previous section.
+
+#### Use a cluster-wide API
+
+We could add a field to the Infrastructure configuration
+resource. Many of our operators, including OLM-based operators,
+already consume the Infrastructure API.
+
+#### Use an operator-specific API
+
+If we choose to implement a new operator to manage scaling for the
+control plane operators (see below), it could define a new API in a
+CRD it owns.
+
+Updating existing operators to read the new field and scale their
+operands and documenting an entirely new API might be more work than
+using the Infrastructure API that would only be useful if the operator
+responding to the API was an add-on.
+
+### Dynamic vs. Static Setting
+
+It is not likely that users would need to change the setting after
+building a cluster, so we could have the installer set a status field
+in whatever CRD we choose for the API. If we do need to support
+scaling later, that work can be described in another enhancement.
+
+### Managing settings for operators installed by cluster-version-operator
+
+The alternatives in this subsection cover the need to change the
+settings for the static manifests currently managed by the
+cluster-version-operator. These approaches would only apply to the
+operators themselves, and not their operands.
+
+#### Use a cluster profile to change the resource requests for control plane operators
+
+We could create a separate cluster profile with different resource
+requests for our control plane components.
+
+This would be quite a lot of work to implement and test.
+
+We may eventually need different profiles for different customers,
+further multiplying the amount of work.
+
+As an organization, we are trying to keep the number of profiles
+small as a general rule.
+
+Profile settings only apply to the cluster operators Deployments, and
+changing the resource requests in the static manifests does not signal
+to the operators that they need to change the settings for their operands.
+
+#### Lower the resource requests for cluster operators statically in the default configuration
+
+We could apply changes to the default resource requests in the static
+manifests used by all deployments, instead of adding one or more new
+cluster profiles.
+
+This would affect all clusters, and if we are too aggressive with the
+changes we may lower values too much and cause important cluster
+operators to be starved for resources, triggering degraded cluster
+performance.
+
+The static settings only apply to the cluster operators Deployments,
+and changing them does not signal to the operators that they need to
+change the settings for their operands.
+
+#### Have a new operator in the release payload perform the scaling for control plane operators
+
+It would use the new API field to determine how to scale (see above).
```

Comment:
sort of ambiguous in the current enhancement is whether the requested scaling is:

> Single-node workloads are smaller, so control-plane consumption is smaller, and we'd like the requests reduced to reflect that.

or if it is:

> Single-node workloads are smaller, so control-plane consumption is smaller.  But it's not smaller enough.  So we need an extra knob to say "hey, control-plane components, we need to to be extra lean on CPU, so feel free to swap in some long poll durations, add heavy internal rate-limiting, and so on to get your consumption down.  We'll accept the decreased performance".

Do we know which of those it is yet?  If we do, can we make it explicit in the `Proposal` and (non-)goal sections?

Reply:
It is definitely the latter. Actual use is less, but the requests have to fit into less space than they do right now (4 CPU instead of 7).

Reply:
I updated the text of the first goal and added a new non-goal to explain this. I'm holding off on writing the proposal section until there's more feedback about the various options listed in the alternatives section.

Reply:
New (non-)goal text looks good to me.  Feel free to mark this thread resolved.